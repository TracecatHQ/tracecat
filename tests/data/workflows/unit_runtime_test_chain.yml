title: Forward chain with runtime tests
#    A -> B -> C
description: |
  Tests correctness, templates + type casting, context passing, tests.
  A and C get overridden by tests
entrypoint: a
actions:
  - ref: a
    action: core.transform.forward
    args:
      value:
        one: 1
        two: 2
        three: 3

  - ref: b
    action: core.transform.forward
    depends_on:
      - a
    args:
      value:
        one: ${{ ACTIONS.a.result.one -> int }}
        two: ${{ ACTIONS.a.result.two -> int }}
        three: ${{ ACTIONS.a.result.three -> int }}

  - ref: c
    action: core.transform.forward
    depends_on:
      - b
    args:
      value:
        one: ${{ ACTIONS.b.result.one }}
        two: ${{ ACTIONS.b.result.two }}
        three: ${{ ACTIONS.b.result.three }}

tests:
  - ref: a
    success:
      one: 111
      two: 222
      three: 333

  - ref: c
    # NOTE: REturn type consistency not enforced
    success: tests/data/workflows/unit_runtime_test_chain_data.json
