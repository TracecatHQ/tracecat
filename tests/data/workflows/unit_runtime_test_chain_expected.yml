ACTIONS:
  a:
    result:
      one: 111
      two: 222
      three: 333
    result_typename: dict
  b:
    result:
      one: 111
      two: 222
      three: 333
    result_typename: dict
  c:
    result:
      hello: world
      foo: bar
    result_typename: dict
INPUTS:

TRIGGER:
